executor/instruction/mod.rs

```diff
@@ -503,404 +515,43 @@ impl<T> Checker<T> {
         context: &ExecutionContext<impl ReadableSnapshot + 'static>,
         row: &MaybeOwnedRow<'_>,
     ) -> Box<FilterFn<T>> {
-        type BoxExtractor<T> = Box<dyn for<'a> Fn(&'a T) -> VariableValue<'a>>;
         let mut filters: Vec<Box<dyn Fn(&T) -> Result<bool, Box<ConceptReadError>>>> =
             Vec::with_capacity(self.checks.len());
 
         for check in &self.checks {
-            match check {
-                &CheckInstruction::Iid { var, iid } => {
-                    let maybe_var_extractor = self.extractors.get(&var);
-                    let var: BoxExtractor<T> = match maybe_var_extractor {
-                        Some(&subtype) => Box::new(subtype),
-                        None => make_const_extractor(&CheckVertex::Variable(var), row, context),
```

Comment:
General question:
I see row being passed to `make_const_extractor`. 
Does that actually need the row?
And in that case, does this function need the row? And does it have to be called in `*Executor::get_iterator` or can it be called in `*Executor::new`

Reply:
Isn't it strange to call something that comes from the row a constant? Or is it like some value with a type attached?